analyze-build warnings

<!-- :warning: Please, try to follow the template -->

### Information

- **RTI Product**: Connext DDS
- **Version**: 6.1.0
- **Operating system**: Linux
- **Compiler**: GCC
- **Compiler version**: 7.5.0
- **Additional information**:


### What is the current behavior?

There are a few warnings during static analysis:

```plaintext
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/custom_transport/c/FileTransport.c:1504:5: warning: Dereference of null pointer (loaded from variable 'dest_address_in')
analyze-build: INFO:     NDDS_Transport_Address_copy(&sendResourceStruct->_address, dest_address_in);
analyze-build: INFO:     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: /home/mnunez/rti_connext_dds-6.1.0/include/ndds/transport/transport_common_impl.h:68:15: note: expanded from macro 'NDDS_Transport_Address_copy'
analyze-build: INFO:    ( *(dst) = *(src) )
analyze-build: INFO:               ^~~~~~
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/custom_transport/c/FileTransport.c:1553:14: warning: Null pointer passed as an argument to a 'nonnull' parameter
analyze-build: INFO:         || (!NDDS_Transport_Address_is_equal(
analyze-build: INFO:              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: /home/mnunez/rti_connext_dds-6.1.0/include/ndds/transport/transport_common_impl.h:63:7: note: expanded from macro 'NDDS_Transport_Address_is_equal'
analyze-build: INFO:    ( !RTIOsapiMemory_compare(l, r, sizeof(NDDS_Transport_Address_t)) )
analyze-build: INFO:       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: /home/mnunez/rti_connext_dds-6.1.0/include/ndds/osapi/osapi_bufferUtils_impl.h:44:30: note: expanded from macro 'RTIOsapiMemory_compare'
analyze-build: INFO:         (((size) == 0) ? 0 : memcmp(l, r, (size_t) (size)))
analyze-build: INFO:                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
analyze-build: INFO: 2 warnings generated.
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx:162:45: warning: Division by zero
analyze-build: INFO:     std::cout << "Avg. pixel: (" << red_sum / pixel_count << ", "
analyze-build: INFO:                                     ~~~~~~~~^~~~~~~~~~~~~
analyze-build: INFO: 1 warning generated.
analyze-build: INFO: /home/mnunez/rticonnextdds-examples/examples/connext_dds/logging_config/c/logging_publisher.c:94:27: warning: Value stored to 'send_period' during its initialization is never read
analyze-build: INFO:     struct DDS_Duration_t send_period = {4,0};
analyze-build: INFO:                           ^~~~~~~~~~~   ~~~~~
analyze-build: INFO: 1 warning generated.
```

### Steps to reproduce the issue

Run `resources/ci_cd/linux_static_analysis.py`.

--- a/examples/connext_dds/custom_transport/c/FileTransport.c
+++ b/examples/connext_dds/custom_transport/c/FileTransport.c
@@ -1499,6 +1499,7 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
 
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {
+        printf("The destination address cannot be null.");
         return 0;
     }
 
@@ -1553,6 +1554,7 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
 
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {
+        printf("The destination address cannot be null.");
         return 0;
     }
 

--- a/examples/connext_dds/custom_transport/c/FileTransport.c
+++ b/examples/connext_dds/custom_transport/c/FileTransport.c
@@ -1549,7 +1549,7 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
 
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {
-        printf("The destination address cannot be null.");
+        printf("WARNING: The destination address cannot be null.");
         return 0;
     }
 

--- a/examples/connext_dds/custom_transport/c/FileTransport.c
+++ b/examples/connext_dds/custom_transport/c/FileTransport.c
@@ -1497,6 +1497,11 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
         return 0;
     }
 
+    /* Check if dest_address_in is null */
+    if (dest_address_in == NULL) {
+        return 0;
+    }
+
     sendResourceStruct = (struct NDDS_Transport_SendResource_FILE *) calloc(
             1,
             sizeof(*sendResourceStruct));
@@ -1543,6 +1548,11 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
 
     /* multicast not supported */
     if (NDDS_Transport_Address_is_multicast(dest_address_in)) {
+        return 0;
+    }
+
+    /* Check if dest_address_in is null */
+    if (dest_address_in == NULL) {
         return 0;
     }
 

--- a/examples/connext_dds/custom_transport/c/FileTransport.c
+++ b/examples/connext_dds/custom_transport/c/FileTransport.c
@@ -1553,6 +1553,11 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
         return 0;
     }
 
+    /* multicast not supported */
+    if (NDDS_Transport_Address_is_multicast(dest_address_in)) {
+        return 0;
+    }
+
     /* This transport can only share a NDDS_Transport_SendResource_t if the
      * destination address and port number match
      */

--- a/examples/connext_dds/custom_transport/c/FileTransport.c
+++ b/examples/connext_dds/custom_transport/c/FileTransport.c
@@ -1502,15 +1502,12 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
         return 0;
     }
 
-<<<<<<< HEAD
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {
         printf("The destination address cannot be null.");
         return 0;
     }
 
-=======
->>>>>>> f7d1a3199231227d1b27f1a9a7970bcef72db7b2
     sendResourceStruct = (struct NDDS_Transport_SendResource_FILE *) calloc(
             1,
             sizeof(*sendResourceStruct));

--- a/examples/connext_dds/custom_transport/c/FileTransport.c
+++ b/examples/connext_dds/custom_transport/c/FileTransport.c
@@ -1484,6 +1484,7 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
 
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {
+        printf("WARNING: The destination address cannot be null.");
         return 0;
     }
 
@@ -1499,12 +1500,6 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
                     me->_property.trace_level);
         }
 
-        return 0;
-    }
-
-    /* Check if dest_address_in is null */
-    if (dest_address_in == NULL) {
-        printf("The destination address cannot be null.");
         return 0;
     }
 
@@ -1551,11 +1546,6 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
 
     struct NDDS_Transport_SendResource_FILE *sendResourceStruct =
             (struct NDDS_Transport_SendResource_FILE *) *sendresource_in;
-
-    /* Check if dest_address_in is null */
-    if (dest_address_in == NULL) {
-        return 0;
-    }
 
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {

--- a/examples/connext_dds/custom_transport/c/FileTransport.c
+++ b/examples/connext_dds/custom_transport/c/FileTransport.c
@@ -1484,7 +1484,7 @@ static RTI_INT32 NDDS_Transport_FILE_create_sendresource_srEA(
 
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {
-        printf("WARNING: The destination address cannot be null.");
+        printf("WARNING: The destination address cannot be NULL.");
         return 0;
     }
 
@@ -1549,7 +1549,7 @@ static RTI_INT32 NDDS_Transport_FILE_share_sendresource_srEA(
 
     /* Check if dest_address_in is null */
     if (dest_address_in == NULL) {
-        printf("WARNING: The destination address cannot be null.");
+        printf("WARNING: The destination address cannot be NULL.");
         return 0;
     }
 

--- a/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
+++ b/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
@@ -147,7 +147,7 @@ bool print_average_pixel_fast(const CameraImage &sample)
     unsigned int pixel_count = pixels.element_count();
 
     if (pixel_count == 0) {
-        print("The number of pixels should be greater than 0.");
+        printf("The number of pixels should be greater than 0.");
         return false;
     }
 

--- a/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
+++ b/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
@@ -147,7 +147,7 @@ bool print_average_pixel_fast(const CameraImage &sample)
     unsigned int pixel_count = pixels.element_count();
 
     if (pixel_count == 0) {
-        printf("The number of pixels should be greater than 0.");
+        std::cout << "WARNING: the pixels count is 0" << std::endl;
         return false;
     }
 

--- a/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
+++ b/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
@@ -147,6 +147,7 @@ bool print_average_pixel_fast(const CameraImage &sample)
     unsigned int pixel_count = pixels.element_count();
 
     if (pixel_count == 0) {
+        print("The number of pixels should be greater than 0.");
         return false;
     }
 

--- a/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
+++ b/examples/connext_dds/flat_data_api/c++98/CameraImage_subscriber.cxx
@@ -146,6 +146,10 @@ bool print_average_pixel_fast(const CameraImage &sample)
 
     unsigned int pixel_count = pixels.element_count();
 
+    if (pixel_count == 0) {
+        return false;
+    }
+
     PixelPlainHelper *pixel_array = rti::flat::plain_cast(pixels);
     if (pixel_array == NULL) {
         return false;

--- a/examples/connext_dds/logging_config/c/logging_publisher.c
+++ b/examples/connext_dds/logging_config/c/logging_publisher.c
@@ -91,7 +91,6 @@ int publisher_main(int domainId, int sample_count)
     DDS_InstanceHandle_t instance_handle = DDS_HANDLE_NIL;
     const char *type_name = NULL;
     int count = 0;
-    struct DDS_Duration_t send_period = {4,0};
     struct DDS_Time_t sourceTimestamp = {0, 0};
 
     /* To customize participant QoS, use

